### PR TITLE
[mail] Add mail content processing

### PR DESCRIPTION
**Pull request by J-N-K**
Originally contributed as https://github.com/smarthomej/addons/pull/131 and https://github.com/smarthomej/addons/pull/159.

This adds content processing to the IMAP and POP3 things. The content can be pre.processed before it is written to a channel.

Signed-off-by: Jan N. Klug <github@klug.nrw>

### CODE REVIEW

File: bundles/org.openhab.binding.mail/README.md
Comment (clinique) [inline bundles/org.openhab.binding.mail/README.md:99]: Mispelled `String` I guess

File: tools/static-code-analysis/checkstyle/suppressions.xml
Comment (clinique) [inline tools/static-code-analysis/checkstyle/suppressions.xml:22]: Is this file supposed to be present ?

**Review by clinique — CHANGES_REQUESTED** [review]

File: tools/static-code-analysis/checkstyle/suppressions.xml
Comment (J-N-K) [reply tools/static-code-analysis/checkstyle/suppressions.xml:22]: This is to suppress false positives from the code-checker.

**Comment by J-N-K** [comment]
@openhab/add-ons-maintainers This is now ready for review after core changes have been merged.

**Review by kaikreuzer — APPROVED** [review]
Lgtm, thanks.
@clinique There is still a veto from you on this PR - is this resolved now? Then please approve.

File: tools/static-code-analysis/checkstyle/suppressions.xml
Comment (clinique) [reply tools/static-code-analysis/checkstyle/suppressions.xml:22]: Ok. - fine for me

**Pull request merged**
